app/src/org/commcare/activities/LoginActivity.java

```diff
@@ -542,6 +546,7 @@ protected void deliverResult(LoginActivity receiver,
                             Toast.makeText(receiver,
                                     Localization.get("login.update.install.success"),
                                     Toast.LENGTH_LONG).show();
+                            DataChangeLogger.log(new DataChangeLog.CommCareAppUpdated());

```

Comment:
@shubham1g5 This actually isn't the code path for an auto update; this is for the demo user restore feature. Are you trying to just log whenever an app update completes? If so, you should be able to do that somewhere in `UpdateActivity` or `UpdateTask`

Reply:
@amstone326 Hey. Would you be able to verify this again. The `deliverResult` here is part of `InstallStagedUpdateTask`. Is not this the task resposible for auto update ?

Reply:
Oh wow I absolutely read this wrong and too quickly, I think because of the way Github collapses the file, it looked like this was coming after line 207/211 above. Sorry about that!!

Reply:
However, this is still not the only place you want to be adding this logic. `InstallStagedUpdateTask` is also launched from `UpdateActivity::launchUpdateInstallTask`

Reply:
That [code path](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/activities/UpdateActivity.java#L439) we already covered while implementing data change log, though this auto update one got left behind.